lib/src/matchers/jest_dom/is_checked.dart

```diff
@@ -99,7 +97,7 @@ class _IsChecked extends Matcher {
     if (item is InputElement) {
       final type = item.getAttribute('type');
       if (_validTypes.contains(type)) {
-        return item.checked;
+        return item.checked!;
```

Comment:
Is this safe if we've only tightened to `InputElement` and not something more specific like `CheckboxInputElement`?

Reply:
It's only usage is right after `isElementThatCanBeChecked` short-circuits for non-checkable elements so I think it is safe. But I'm going to move the `isElementThatCanBeChecked` into this function in case the method is ever used in multiple places